src/librustc_resolve/diagnostics.rs

```diff
@@ -397,6 +397,102 @@ impl Bar {
 ```
 "##,
 
+E0412: r##"
+An undeclared type name was used. Example of erroneous codes:
+
+```
+impl Something {} // error: use of undeclared type name `Something`
+// or:
+trait Foo {
+    fn bar(N); // error: use of undeclared type name `N`
+}
+```
+
+To fix this error, please verify you didn't misspell the type name or
+you did declare it. Examples:
+
+```
+struct Something;
+
+impl Something {}
+// or:
+trait Foo {
+    type N;
+
+    fn bar(Self::N);
+}
+```
+"##,
+
+E0413: r##"
+A declaration shadows an enum variant or unit-like struct in scope.
+Example of erroneous code:
+
+```
+struct Foo;
+
+let Foo = 12i32; // error: declaration of `Foo` shadows an enum variant or
+                 //        unit-like struct in scope
+```
+
+
+To fix this error, you have to change the name of the declaration of the
```

Comment:
You suggest changing name of the shadowee, but in example you change the shadower.